src/com/oltpbenchmark/benchmarks/tpcc/JsonMetricsHelper.java

```diff
@@ -0,0 +1,229 @@
+                    "\nError Message:" + e.getMessage());
+        }
+        String dest = currentDir + File.separator + outputDirectory + File.separator + "output.json";
+
+        try (FileWriter fw = new FileWriter(dest)) {
+            new GsonBuilder().setPrettyPrinting().create().toJson(tpccRunResults, fw);
+        } catch (IOException e) {
+            LOG.error("Got exception while writing JSON metrics to file.");
+            e.printStackTrace();
+            return;
+        }
+        LOG.info("Output json data into file: " + dest);
+    }
+
+    private static double computeAverage(double old_value, double newValue, int count) {
+        return ((old_value * count) + newValue) / (count + 1);
+    }
+
+    /*
+    Will merge the results from json files in the given directory
+     */
+    public static void mergeJsonResults(String dirPath) {
+        List<TpccRunResults> listTpccRunResults = new ArrayList<>();
+        TpccRunResults mergedTpccResults = new TpccRunResults();
+        String[] fileNames = new File(dirPath).list();
+        Gson gson = new Gson();
+        for (String file : fileNames) {
+            if (!file.endsWith("json")) {
+                continue;
+            }
+            try {
+                listTpccRunResults.add(gson.fromJson(
+                        new String(Files.readAllBytes(Paths.get(file))), TpccRunResults.class));
+            } catch (IOException ie) {
+                LOG.error("Got exception while reading json file - " + file + " : ", ie);
+                return;
+            }
+        }
+        int numNewOrder = 0;
+
+        int fileCnt = 0;
+        for (TpccRunResults tpccResult : listTpccRunResults) {
+            mergedTpccResults.TestConfiguration = tpccResult.TestConfiguration;
+
+            if(fileCnt == 0) {
+                mergedTpccResults.Results.throughputMin = tpccResult.Results.throughput;
+                mergedTpccResults.Results.throughputMax = tpccResult.Results.throughput;
+            } else {
+                if (mergedTpccResults.Results.throughputMin > tpccResult.Results.throughput)
+                    mergedTpccResults.Results.throughputMin = tpccResult.Results.throughput;
+                if (mergedTpccResults.Results.throughputMax < tpccResult.Results.throughput)
+                    mergedTpccResults.Results.throughputMax = tpccResult.Results.throughput;
+            }
+            mergedTpccResults.Results.throughput = computeAverage(mergedTpccResults.Results.throughput,
+                    tpccResult.Results.throughput, fileCnt);
+
+            List<TpccRunResults.LatencyList> latList = tpccResult.Latencies;
+            for (int i = 0; i < latList.size() ; i++) {
+                TpccRunResults.LatencyList opLatency = latList.get(i);
+                TpccRunResults.LatencyList latency;
+                if(opLatency.Transaction.equalsIgnoreCase("NewOrder"))
+                    numNewOrder += opLatency.Count;
```

Comment:
this should only include *successful* new order operations, right?

Reply:
yes, this is only the successful new order transaction counts.